server/events/vcs/gitea/client.go

```diff
@@ -0,0 +1,481 @@
+// Copyright 2024 Martijn van der Kleijn & Florian Beisel
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitea
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"time"
+
+	"code.gitea.io/sdk/gitea"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/logging"
+)
+
+// Emergency break for Gitea pagination (just in case)
+// Set to 500 to prevent runaway situations
+// Value chosen purposely high, though randomly.
+const giteaPaginationEBreak = 500
+
+type GiteaClient struct {
+	giteaClient *gitea.Client
+	username    string
+	token       string
+	pageSize    int
+	ctx         context.Context
+	logger      logging.SimpleLogging
```

Comment:
You don't need this `logger` property any more.

Reply:
Good point, removed.